python/ray/services.py

```diff
@@ -892,32 +895,39 @@ def start_dashboard(redis_address,
     dashboard_filepath = os.path.join(
         os.path.dirname(os.path.abspath(__file__)), "dashboard/dashboard.py")
     command = [
-        sys.executable, "-u", dashboard_filepath,
+        sys.executable,
+        "-u",
+        dashboard_filepath,
         "--redis-address={}".format(redis_address),
         "--http-port={}".format(port),
         "--token={}".format(token),
+        "--temp-dir={}".format(temp_dir),
     ]
     if redis_password:
         command += ["--redis-password", redis_password]
 
+    if sys.version_info <= (3, 0):
+        return None, None
     try:
-        process_info = start_ray_process(
-            command,
-            ray_constants.PROCESS_TYPE_DASHBOARD,
-            stdout_file=stdout_file,
-            stderr_file=stderr_file)
-    except Exception:
-        logger.warning("Failed to start the dashboard. The dashboard requires "
-                       "python3 and 'pip install psutil setproctitle'.")
-    else:
-        dashboard_url = ("http://localhost:{}/?token={}".format(
-            port, token))
-        print("\n" + "=" * 70)
-        print("View the dashboard at {}".format(dashboard_url))
-        print("=" * 70 + "\n")
-        return dashboard_url, process_info
+        import aiohttp  # noqa: F401
+        import psutil  # noqa: F401
+    except ImportError:
+        logger.warning(
+            "Failed to start the dashboard. The dashboard requires Python 3 "
+            "as well as 'pip install aiohttp psutil'.")
+        return None, None
 
-    return None, None
+    process_info = start_ray_process(
+        command,
+        ray_constants.PROCESS_TYPE_DASHBOARD,
+        stdout_file=stdout_file,
+        stderr_file=stderr_file)
+    dashboard_url = "http://{}:{}/?token={}".format(
```

Comment:
I'm not sure why, but I can't seem to connect to the dashboard using `localhost` or `127.0.0.1`. I have to use the value returned by `ray.services.get_node_ip_address()`.

Reply:
In `dashboard.py` we explicitly bind to that IP.

Should just bind to 0.0.0.0. Line 184.